### ISSUE TITLE
generate swagger.json on build 

### ISSUE THREAD

**Issue by DmytroKostin0** [inaka/cowboy_swagger#169]
is there any way to generate swagger.json on build time?

**Comment by paulo-ferraz-oliveira** [comment]
Not at this moment, I don't think. It shouldn't be complex to implement, as a hook on `cowboy_swagger:to_json/1`, but I'm not sure we'll want that. @elbrujohalcon?

**Comment by elbrujohalcon** [comment]
Yeah… I would say it should not be _the default app behaviour_. But, if it's clearly optional, we can accept a PR with such changes.

**Comment by paulo-ferraz-oliveira** [comment]
👍 then I propose a behaviour with a single function `cowboy_swagger_json_exporter:export/1` where the input is the generated `swagger.json`. Then with an application config. key like e.g. `json_exporter_mod`, you can add a module name that'll be called when the JSON is done.

**Comment by elbrujohalcon** [comment]
Sounds good to me
